### DISCUSSION TITLE
platform/integrations/langchain

### DISCUSSION THREAD

**Discussion by giscus** [apify/apify-docs#930] [Comments]
# platform/integrations/langchain

Learn how to integrate Apify with 🦜🔗 LangChain, in order to feed vector databases and LLMs with data crawled from the web.

https://docs.apify.com/platform/integrations/langchain

**Comment by JackaDu** [comment]
getting the following error:
from langchain.document_loaders.base import Document
ImportError: cannot import name 'Document' from 'langchain.document_loaders.base' (/Users/jackdu/.local/share/virtualenvs/apify_langchain-rKxSQDZS/lib/python3.9/site-packages/langchain/document_loaders/base.py)

**Reply by jirispilka** [reply]
@JackaDu thank you for pointing this out. Langchain package structure was changed and the `Document` was moved to langchain_core.

The documentation will be soon updated, please check this PR for now: https://github.com/apify/apify-docs/pull/934

Thanks again!

**Comment by TanviBhardwaj24** [comment]
For this line in the tutorial, index = VectorstoreIndexCreator().from_loaders([loader]), I am getting the following error: line 341, in __init__
    raise validation_error
pydantic.v1.error_wrappers.ValidationError: 1 validation error for VectorstoreIndexCreator
embedding
  field required (type=value_error.missing)

**Reply by jirispilka** [reply]
@TanviBhardwaj24 thank you! Good catch.

Langchain has recently changed `VectorstoreIndexCreator` and OpenAI is no longer a default provider and needs to be passed explicitly. 

```python
index = VectorstoreIndexCreator(embedding=OpenAIEmbeddings()).from_loaders([loader])

result = index.query_with_sources(query, llm=OpenAI())
```

The updated documentation should be published within a day or two. You can check this PR for now: https://github.com/apify/apify-docs/pull/1051

**Reply by TanviBhardwaj24** [reply]
Thank you so much, it works now!
